src/frontend/src/scheduler/distributed/query.rs

```diff
@@ -525,15 +523,10 @@ mod tests {
 
     fn generate_parallel_units(start_id: u32, node_id: u32) -> Vec<ParallelUnit> {
         let parallel_degree = 8;
-        let mut parallel_units = vec![ParallelUnit {
-            id: start_id,
-            r#type: ParallelUnitType::Single as i32,
-            worker_node_id: node_id,
-        }];
-        for id in start_id + 1..start_id + 1 + parallel_degree {
+        let mut parallel_units = vec![];
+        for id in start_id..start_id + parallel_degree {
```

Comment:
Or `Vec::with_capacity`.
```suggestion
        (start_id..start_id + parallel_degree).map(..).collect()
```